### PR TITLE
Set default values in a better way

### PR DESCRIPTION
**Pull request by bording**
@timbussmann This allows for any of the properties to be overridden by just setting the value in the project file, no special properties or conventions to worry about.

I've also added `GenerateDocumentationFile` to the list of properties we set by default, since there's not much point in generating the xml docs file if you aren't going to package it.

I've also added a default Description value (setting it to the package id) so that's not really required now either. I didn't update the README about that yet, though.

### CODE REVIEW

File: src/Particular.Packaging/build/Particular.Packaging.targets [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
 
   <PropertyGroup>
-    <Authors>Particular Software</Authors>
-    <Company>NServiceBus Ltd</Company>
-    <PackageLicenseUrl Condition="'$(PackageLicenseUrl)' == ''">http://particular.net/LicenseAgreement</PackageLicenseUrl>
-    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
-    <Copyright>© 2010-$([System.DateTime]::UtcNow.ToString(yyyy)) NServiceBus Ltd. All rights reserved.</Copyright>
-    <PackageTags Condition="'$(PackageTags)' == ''">nservicebus messaging</PackageTags>
-    <PackageIconUrl Condition="'$(PackageIconUrl)' == ''">http://s3.amazonaws.com/nuget.images/NServiceBus_32.png</PackageIconUrl>
+    <PackageId Condition="'$(PackageId)' == ''">$(AssemblyName)</PackageId>
```
Comment (timbussmann) [inline src/Particular.Packaging/build/Particular.Packaging.targets]: so this remains in the targets because the `AssemblyName` isn't set when the prop file executed?

File: src/Particular.Packaging/build/Particular.Packaging.props [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<Project>
+
+  <PropertyGroup>
+    <Authors>Particular Software</Authors>
```
Comment (timbussmann) [inline src/Particular.Packaging/build/Particular.Packaging.props]: so, just make sure these elements get overridden later by custom defined elements?
(this could however break in case some project decides to set their nuget properties in the Directory.Build.props, doesn't it?)

File: .gitignore [hunk 3]
```diff
@@ -60,7 +60,6 @@ installer/ServiceControl-cache
 [Rr]eleases/
 x64/
 x86/
-build/
```
Comment (timbussmann) [inline .gitignore]: why is this needed?

File: src/Particular.Packaging/build/Particular.Packaging.props (hunk 2 above)
Comment (bording) [reply src/Particular.Packaging/build/Particular.Packaging.props]: I went back and double checked this, and it looks like what I stated in our conversation was wrong. The general order of evaluation is:

- Directory.Build.props
- NuGet package props files
- Project file
- NuGet package targets files
- Directory.Build.targets

So there is potentially an ordering problem here. I'm going to think about the ramifications of this a bit, and might make some additional PR changes.

File: src/Particular.Packaging/build/Particular.Packaging.targets (hunk 1 above)
Comment (bording) [reply src/Particular.Packaging/build/Particular.Packaging.targets]: Correct

File: .gitignore (hunk 3 above)
Comment (bording) [reply .gitignore]: To prevent the newly added `Particular.Packaging.props` file from being ignored.

**Comment by bording** [comment]
@timbussmann With the changes I just pushed, adding settings to Directory.Build.props should work as well.

Unless you've got any more questions, this should be ready now!
